File: app/src/main/java/com/tuhoang/pocketmind/utils/AppLogger.java
package com.tuhoang.pocketmind.utils;

import android.util.Log;

public class AppLogger {
    private static final String DEFAULT_TAG = "PocketMindLog";

    public static void d(String tag, String message) {
        Log.d(tag, message);
    }

    public static void d(String message) {
        Log.d(DEFAULT_TAG, message);
    }

    public static void e(String tag, String message, Throwable t) {
        // Funnel all error logs here to avoid raw e.printStackTrace() and potential data leaks
        Log.e(tag, message, t);
    }

    public static void e(String tag, String message) {
        Log.e(tag, message);
    }
}

File: app/src/main/kotlin/com/tuhoang/pocketmind/utils/AppLogger.kt
package com.tuhoang.pocketmind.utils

import android.util.Log

object AppLogger {
    private const val DEFAULT_TAG = "PocketMindLog"

    fun d(tag: String, message: String) = Log.d(tag, message)
    fun d(message: String) = Log.d(DEFAULT_TAG, message)
    fun e(tag: String, message: String, t: Throwable? = null) = Log.e(tag, message, t)
}
